lib/matplotlib/markers.py

```diff
@@ -229,7 +230,9 @@ def set_fillstyle(self, fillstyle):
             raise ValueError("Unrecognized fillstyle %s"
                              % ' '.join(self.fillstyles))
         self._fillstyle = fillstyle
-        self._recache()
+        # do not call _recache() if marker is not yet set up
+        if self._marker is not None:
```

Comment:
Maybe add a `_defer_recache=False` kwarg to both `set_marker` and `set_fillstyle`?  That way in `__init__` you can prevent it from running.

A more invasive (but maybe better?) solution could be to use the stale/invalid pattern and then in all of the `get_*` methods check if it is stale and if so recache it.